components/List/List.js

```diff
@@ -1,16 +1,30 @@
-import { StyledList, StyledItem } from "@/styles";
+import { StyledList, StyledItem, StyledDeleteButton } from "@/styles";
+
+export default function List({ transactions, setTransactions }) {
+  function deleteTransaction(id) {
+    const updatedTransactions = transactions.filter(
+      (transaction) => transaction.id !== id
+    );
+    setTransactions(updatedTransactions);
+  }
 
-export default function List({ transactions }) {
   return (
     <>
       <StyledList>
-        {transactions.map((transaction) => (
-          <StyledItem key={transaction.id}>
-            {transaction.date} - {transaction.amount} EUR -{" "}
-            {transaction.category} - {transaction.type} -{" "}
-            {transaction.description}
-          </StyledItem>
-        ))}
+        <>
+          {transactions.map((transaction) => (
+            <StyledItem key={transaction.id}>
+              {transaction.date} - {transaction.amount} EUR -{" "}
+              {transaction.category} - {transaction.type} -{" "}
+              {transaction.description}{" "}
+              <StyledDeleteButton
+                onClick={() => deleteTransaction(transaction.id)}
+              >
+                ❌
+              </StyledDeleteButton>
+            </StyledItem>
+          ))}
```

Comment:
```suggestion
  {transactions.map(({ id, date, amount, category, type, description }) => (
    <StyledItem key={id}>{date} - {amount} EUR - {category} - {type} - {description}
    <StyledDeleteButton onClick={() => deleteTransaction(id)}>
      ❌
    </StyledDeleteButton>
  </StyledItem>
))
}
```
I think using the destructuring syntax here might be useful & help clean up the code a bit.